ma_gym/envs/lumberjacks/lumberjacks.py

```diff
@@ -0,0 +1,351 @@
+import copy
+import itertools
+import logging
+import random
+from dataclasses import dataclass
+from typing import List, Mapping, Tuple, Union
+
+import gym
+import numpy as np
+from gym import spaces
+from gym.utils import seeding
+from PIL import Image, ImageColor, ImageDraw
+
+from ..utils.action_space import MultiAgentActionSpace
+from ..utils.draw import draw_circle, draw_grid, fill_cell, write_cell_text
+from ..utils.observation_space import MultiAgentObservationSpace
+
+logger = logging.getLogger(__name__)
+
+Coordinates = Tuple[int, int]
+
+
+@dataclass
+class Agent:
+    """Dataclass keeping all data for one agent/lumberjack in environment.
+
+    Attributes:
+        id: unique id in one environment run
+        pos: position of the agent in grid
+    """
+    id: int
+    pos: Coordinates
+
+
+class Lumberjacks(gym.Env):
+    """
+    Lumberjacks environment involve a grid world, in which multiple lumberjacks attempt to cut down all trees. In order to cut down a tree in given cell, there must be present greater or equal number of agents/lumberjacks then the tree strength in the same location as tree. Tree is then cut down automatically.
+
+    Agents select one of fire actions ∈ {No-Op, Down, Left, Up, Right}.
+    Each agent's observation includes its:
+        - agent ID (1)
+        - position with in grid (2)
+        - number of steps since beginning (1)
+        - number of agents and tree strength for each cell in agent view (2 * `np.prod(tuple(2 * v + 1 for v in agent_view))`).
+    All values are scaled down into range ∈ [0, 1].
+
+    Only the agents who are involved in cutting down the tree are rewarded with `tree_cutdown_reward`.
+    The environment is terminated as soon as all trees are cut down.
+
```

Comment:
Also, add the following line:

"Upon rendering, we show the grid, where each cell shows the agent( blue) and/or tree( green) with their current strength."

This is particularly different than other env.s because we show the id of agents in the other environment to identify their location.